modules/platforms/dotnet/Apache.Ignite.Core/Cache/Configuration/QueryEntity.cs

```diff
@@ -229,12 +229,31 @@ string IQueryEntityInternal.GetAlias(string fieldName)
             return _aliasMap.TryGetValue(fieldName, out res) ? res : null;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QueryEntity"/> class.
+        /// </summary>
+        /// <param name="reader">The reader.</param>
+        internal QueryEntity(IBinaryRawReader reader)
+        {
+            Read(reader, ClientSocket.CurrentProtocolVersion);
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="QueryEntity"/> class.
         /// </summary>
         /// <param name="reader">The reader.</param>
         /// <param name="srvVer">Server version.</param>
         internal QueryEntity(IBinaryRawReader reader, ClientProtocolVersion srvVer)
+        {
+            Read(reader, srvVer);
+        }
+        
+        /// <summary>
+        /// Reads an instance of the <see cref="QueryEntity"/> class.
+        /// </summary>
+        /// <param name="reader">The reader.</param>
+        /// <param name="srvVer">Server version.</param>
+        internal void Read(IBinaryRawReader reader, ClientProtocolVersion srvVer)
```

Comment:
Ideally we should get rid of `ClientProtocolVersion` here and separate Thin and Thick logic completely. Thin client config logic should be in `ClientCacheConfigurationSerializer`.